### ISSUE TITLE
Blacklist doesn't work?

### ISSUE THREAD

**Issue by a-torgovitsky** [ntpeters/vim-better-whitespace#68]
I'm having trouble getting the blacklist feature to work.
The only option I have added to my .vimrc is:
```
autocmd BufEnter * EnableStripWhitespaceOnSave
```
As expected, for most files trailing whitespaces are removed on save, without me needing to turn on the feature. However, what is unexpected is that this is also the case for Markdown files, which are supposed to be in the default blacklist.

I have tried editing the blacklist by adding
```
let g:better_whitespace_filetypes_blacklist=['markdown']
```
but I still see the same behavior.

After loading a Markdown file, I can turn off the whitespace striping with
```
ToggleStripWhitespaceOnSave
```

Am I interpreting the blacklist incorrectly? Or is there an issue with it? My Vim version is 8.0.710.

Thanks for the great plugin!

**Comment by ntpeters** [comment]
Unfortunately, the blacklist is a bit misleading at the moment. It prevents the plugin from being enabled by default for the blacklisted file types, but it’s possible to manually execute the commands yourself (as you’re seeing here).

I hope to make this behave more as expected in the near future. For now, you should be able to get the desired behavior by updating your auto command to inspect the blacklist itself:

```
autocmd BufEnter * if index(g:better_whitespace_filetypes_blacklist, &ft) >= 0 | EnableStripWhitespaceOnSave | endif
```

Sorry for things being a bit confusing right now. Hope that helps!

**Comment by a-torgovitsky** [comment]
I get an error from that command:

```
Error detected while processing BufEnter Auto commands for "*":
E488: Trailing characters:  EnableStripWhitespaceOnSave | endif
```

No expert on vimscript here, but the syntax looks right, doesn't it?

Closest thing I could find was [this stack exchange discussion](https://stackoverflow.com/questions/36462113/calling-vim-plugin-command-in-autocmd-fails-with-e488-trailing-characters)

**Comment by ntpeters** [comment]
Ah, good find. Does the proposed solution in that post work for you?

```
autocmd BufEnter * if index(g:better_whitespace_filetypes_blacklist, &ft) < 0 | exec 'EnableStripWhitespaceOnSave' | endif
```

**Comment by a-torgovitsky** [comment]
That works -- thanks!

**Comment by dylan-chong** [comment]
@ntpeters Why not just add that line to the plugin itself? That would make a good enough solution for now.

(I guess it doenst take into accont the user opening a non-blacklisted file then switching to a blacklisted one, where whitespace will be stripped on save).

**Comment by ntpeters** [comment]
Fixed in #72
